src/devtools/client/webconsole/components/Input/JSTerm.js

```diff
@@ -59,20 +65,38 @@ class JSTerm extends React.Component {
   }
 
   async componentDidMount() {
-    this.editorWaiter = createEditor({ execute: this.execute });
+    this.editorWaiter = createEditor({
+      execute: this.execute,
+      historyCursorUp: this.historyCursorUp,
+      historyCursorDown: this.historyCursorDown,
+    });
     this.editor = await this.editorWaiter;
     this.editor.appendToLocalElement(this.node);
 
     const recordingId = getRecordingId();
     const recording = await getRecording(recordingId);
 
-    this.setState({ canEval: recording.userRole !== "team-user" });
+    this.setState({ canEval: recording.userRole !== "team-user", historyIndex: 0 });
```

Comment:
Small nit: any reason why we couldn't just set the `historyIndex` in the constructor?